pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -285,6 +273,9 @@ func getListenersByLoadBalancerID(client *gophercloud.ServiceClient, id string)
 		return true, nil
 	})
 	if err != nil {
+		if isNotFound(err) {
```

Comment:
What's the benefit of pushing this down here (and similar elsewhere), rather than have the caller use `isNotFound(...)`?   From experience I'm nervous about ensuring that all functions correctly convert Openstack not-found errors into `ErrNotFound` - we've certainly had places where this was missed in the past.